server/src/instant/db/model/attr.clj

```diff
@@ -257,104 +259,121 @@
+                                                               attr-table-cols))}]]
+                             :join [:ident-ids
+                                    [:= :attr-values.forward-ident :ident-ids.id]]}]
+                           :on-conflict {:on-constraint :attrs_pkey}
+                           :do-update-set {:value_type [:case
+                                                        (list* :and
+                                                               (map (fn [col]
+                                                                      ;; Some fields can be null, so we need to
+                                                                      ;; use "distinct from" instead of "="
+                                                                      [:raw [[:inline (qualify-col :attrs col)]
+                                                                             " is not distinct from "
+                                                                             [:inline (qualify-col :EXCLUDED col)]]])
+                                                                    attr-table-cols))
+                                                        :EXCLUDED.value_type
+                                                        ;; raise_exception_message is typed to return
+                                                        ;; a boolean, so we cast it to text so that it
+                                                        ;; can throw its exception
+                                                        :else [:cast
+                                                               [:raise_exception_message
+                                                                [:||
+                                                                 "The attribute with id "
+                                                                 [:cast :EXCLUDED.id :text]
+                                                                 " conflicts with an existing attribute with id "
+                                                                 [:cast :attrs.id :text] "."]]
+                                                               :text]]}
+                           :returning :*}]]
+                  :select [[[:json_build_object
+                             "idents" [:coalesce
+                                       {:select [[[:json_agg [:row_to_json :ident-inserts]]]]
+                                        :from :ident-inserts}
+                                       [:cast [:inline "[]"] :json]]
+                             "attrs" [:coalesce
+                                      {:select [[[:json_agg [:row_to_json :attr-inserts]]]]
+                                       :from :attr-inserts}
+                                      [:cast [:inline "[]"] :json]]]]]}
+           result (sql/execute-one! ::insert-multi! conn (hsql/format query))]
+       {:attrs (-> result
```

Comment:
Lots of whitespace changes here, but this is the only real change. I need to get the attrs that were created so that we can add nulls for existing entities if we added an indexed blob attr.